### PR TITLE
[ready-for-core-team-review]CRM-19485, Fixed selector issue on Batch Transaction form

### PR DESCRIPTION
**Pull request by pradpnayak**
---
- CRM-19485: Selector issue on Batch trxn assignment page
  https://issues.civicrm.org/jira/browse/CRM-19485

### CODE REVIEW

**Comment by JoeMurray** [comment]
I've tested successfully and reviewed code. Ready to merge.

File: templates/CRM/Financial/Form/BatchTransaction.tpl [hunk 1]
```diff
@@ -238,12 +242,14 @@ function buildTransactionSelectorAssign(filterSearch) {
...
       "dataType": 'json',
       "type": "POST",
+      'async': false,
       "url": sSource,
       "data": aoData,
       "success": fnCallback
     });
   }
 });
+toggleFinancialSelections('#toggleSelect', 'assign');
```
Comment (colemanw) [inline templates/CRM/Financial/Form/BatchTransaction.tpl]: Disabling async in ajax calls is deprecated.
Add this call to your fnCallback instead and you don't need to do so. Like this:

    "success": function() {
      toggleFinancialSelections('#toggleSelect', 'assign');
      fnCallback();
    }

File: templates/CRM/Financial/Form/BatchTransaction.tpl [hunk 2]
```diff
@@ -295,10 +301,12 @@ function buildTransactionSelectorRemove( ) {
       "type": "POST",
       "url": sSource,
       "data": aoData,
+      'async': false,
       "success": fnCallback
     });
   }
 });
+toggleFinancialSelections('#toggleSelects', 'remove');
```
Comment (colemanw) [inline templates/CRM/Financial/Form/BatchTransaction.tpl]: Ditto.

File: templates/CRM/Financial/Form/BatchTransaction.tpl [hunk 3]
```diff
@@ -172,6 +163,19 @@ function enableActions( type ) {
   }
 }
 
+function toggleFinancialSelections(toggleID, toggleClass) {
+  mark = 'x';
```
Comment (colemanw) [inline templates/CRM/Financial/Form/BatchTransaction.tpl]: missing `var` keyword

File: templates/CRM/Financial/Form/BatchTransaction.tpl [hunk 4]
```diff
@@ -52,6 +52,7 @@
             {else}
             <td>&nbsp;</td>
           {/if}
+          </tr>
```
Comment (colemanw) [inline templates/CRM/Financial/Form/BatchTransaction.tpl]: was this missing previously?

**Review by colemanw — CHANGES_REQUESTED** [review]

File: templates/CRM/Financial/Form/BatchTransaction.tpl (hunk 4 above)
Comment (pradpnayak) [inline templates/CRM/Financial/Form/BatchTransaction.tpl]: Yes, it was missing

**Comment by pradpnayak** [comment]
@colemanw thanks for the alternate solution for async. I have updated the PR with the changes. Could you please review?

**Review by colemanw — APPROVED** [review]

**Pull request merged**